app/services/catalog/notify.rb

```diff
@@ -0,0 +1,28 @@
+module Catalog
+  class Notify
+    ACCEPTABLE_NOTIFICATION_CLASSES = %w[OrderItem ApprovalRequest].freeze
+
+    attr_reader :notification_object
+
+    def initialize(klass, id, payload)
+      raise Catalog::InvalidNotificationClass unless ACCEPTABLE_NOTIFICATION_CLASSES.include?(klass.camelcase)
+
+      @notification_object = klass.camelcase.constantize.find(id)
+      @payload = payload
+    end
+
+    def process
+      @notification_object.update(:state => @payload["decision"])
+
+      case @notification_object
+      when OrderItem
+        Catalog::OrderStateTransition.new(@notification_object.order.id).process
+      when ApprovalRequest
+        @notification_object.update(:reason => @payload["reason"])
+        Catalog::ApprovalTransition.new(@notification_object.order_item.id).process
+      end
+
+      self
```

Comment:
Do we need `self` returned? The minion may only need a status `:ok` ( or other error status if there is an error. ) The transition service would handle the logging if there is a problem, so the minion should really just know if the `process` method was able to run without issue.

Reply:
*Need*? Probably not, I was just following convention with all of our other services. Right now I figured we would just return the object. The `state` will be changed, and the transition services do the other state changes for the related objects.